### PR TITLE
Feature Request:Allow threads to be yielded forcefully and externally.

### PR DESCRIPTION
**Pull request by RadiantUwU**
#109

### CODE REVIEW

File: src/classes/luaCoroutine.cpp [hunk 1]
```diff
@@ -182,4 +183,13 @@ bool LuaCoroutine::isDone() {
 int LuaCoroutine::luaYield(lua_State *state) {
     int argc = lua_gettop(state);
     return lua_yield(state, argc);
-}
\ No newline at end of file
+}
+
+void LuaCoroutine::pause_execution() {
+    if (!done) lua_sethook(tState,terminate_hook,LUA_MASKCOUNT,1); // force it to run next time
```
Comment (Trey2k) [inline src/classes/luaCoroutine.cpp:189]:
```suggestion
    if (!done) lua_sethook(tState,terminate_hook, LUA_MASKCOUNT, 1); // force it to run next time
```

File: src/classes/luaCoroutine.cpp [hunk 2]
```diff
@@ -182,4 +183,13 @@ bool LuaCoroutine::isDone() {
...
+}
+
+void LuaCoroutine::terminate_hook(lua_State* tState,lua_Debug* dbg) {
+   lua_sethook(tState,nullptr,NULL,NULL); // remove hook
+   lua_yield(tState,0);
```
Comment (Trey2k) [inline src/classes/luaCoroutine.cpp:194]:
```suggestion
   lua_sethook(tState ,nullptr, NULL, NULL); // remove hook
   lua_yield(tState, 0);
```

**Review by Trey2k — CHANGES_REQUESTED** [review]
Only have a few styling notes. As I am trying to push this code base to more consistent styling in general.

**Comment by RadiantUwU** [comment]
Okay, i did do those changes.

**Review by Trey2k — COMMENTED** [review]
Looks good to me now, I will test changes later this evening.

**Comment by RadiantUwU** [comment]
Got it.

**Comment by Trey2k** [comment]
Again, appreciate the help and feel free to join my [discord](https://discord.gg/QU9kVBXVEa) as well for discussion about the module.

**Review by Trey2k — APPROVED** [review]
This seems to work as intended. I have tested it with the following script
```gdscript
extends Node

var lua: LuaAPI
var coroutine: LuaCoroutine
const killSec := 10


func _ready():
	lua = LuaAPI.new()
	coroutine = lua.new_coroutine()
	coroutine.load_string("
	i = 0
	while true do
		i = i + 1
		if i%100000000 == 0 then
			print('Hello world!')
		end
	end
	")

var running := false
var timeRunning := 0.0
var thread := Thread.new()
func _process(delta):
	if not running:
		thread.start(_thread)
		running = true
	else:
		timeRunning += delta
		if timeRunning >= killSec and thread.is_started():
			coroutine.pause_execution()
			thread.wait_to_finish()
			timeRunning = 0
			running = false


func _thread():
	print("Start")
	var ret = coroutine.resume()
	if ret is LuaError:
		print("ERROR %d: " % ret.type + ret.msg)
		return
	print("Paused")
```

**Pull request merged**